src/main/java/duke/logic/commands/NewItineraryCommand.java

```diff
@@ -50,47 +42,9 @@ public NewItineraryCommand(LocalDateTime start, LocalDateTime end, String hotel,
     public CommandResultText execute(Model model) throws DukeException, FileNotFoundException {
         Venue hotelLocation = ApiParser.getLocationSearch(hotel);
         Itinerary itinerary = new Itinerary(start, end, hotelLocation, name);
-        AgendaList agendaList = new AgendaList();
-        int i = 4;
-        try {
-            while (i < itineraryDetails.length) {
-                List<Venue> venueList = new ArrayList<>();
-                List<Todo> todoList = new ArrayList<>();
-                final int number = Integer.parseInt(itineraryDetails[i++]);
-                while (itineraryDetails[i].equals("/venue")) {
-                    i++;
-                    venueList.add(ApiParser.getLocationSearch(itineraryDetails[i++]));
-                    StringBuilder todos = new StringBuilder();
-                    if (i == itineraryDetails.length - 1 || itineraryDetails[i].matches("-?\\d+")) {
-                        throw new ParseException(Messages.ITINERARY_EMPTY_TODOLIST);
-                    }
-                    todos.append(itineraryDetails[++i]).append("|");
-                    i++;
-                    while (itineraryDetails[i].equals("/and")) {
-                        i++;
-                        todos.append(itineraryDetails[i++]).append("|");
-                        if (i >= itineraryDetails.length) {
-                            break;
-                        }
-                    }
-                    todoList = ParserStorageUtil.getTodoListFromStorage(todos.toString());
-                    if (i >= itineraryDetails.length) {
-                        break;
-                    }
-                }
-                Agenda agenda = new Agenda(todoList, venueList, number);
-                agendaList.add(agenda);
-            }
-        } catch (ArrayIndexOutOfBoundsException e) {
-            throw new ParseException(Messages.ITINERARY_FAIL_CREATION);
-        } catch (NumberFormatException e) {
-            throw new ParseException(Messages.ITINERARY_INCORRECT_COMMAND);
-        }
-        itinerary.setTasks(agendaList);
-
-
-        model.saveItinerary(itinerary);
-        model.itineraryListSave(itinerary);
-        return new CommandResultText("New Itinerary Created :" + itinerary.printItinerary());
+        itinerary.getNumberOfDays();
+        model.storeNewItinerary(itinerary, itineraryDetails);
```

Comment:
For setters, don't use `storeNewItinerary`, use `setNewItinerary`

Reply:
Done